[iantsaii] iP

iantsaii opened the pull request:
# Rocket 
> Do better ~ Anon

Rocket helps you focus on what you need to do, so that you can _do better_
- note down tasks, deadlines, and events
- find tasks by searching for them
- mark tasks as done

All you need to do is,
1. download it from [here](https://github.com/iantsaii/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. **let it manage** your tasks for you 🎉 

Features:
- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

For java `programmers`, you can also use it to practise writing OOP.
```java
package rocket;

public class Command {

    private boolean isExit;
    public Command (boolean isExit) {
        this.isExit = isExit;
    }

    /**
     * Executes the command.
     * @param tasks Modifies or uses the tasklist if needed.
     * @param ui Displays to the UI if needed.
     * @param storage Stores and retrieves data from the storage.
     */
    public void execute(TaskList tasks, Ui ui, Storage storage) {

    }

    /**
     * Tells Rocket if the command is an exit command
     * @return a boolean: true if the command is an exit commnad, false if not
     */
    public boolean isExit() {
        return isExit;
    }
}
```

Modified from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Changed name from Duke to Yello

Commit:
Greet and Exit

Commit:
Renamed chatbot from Yello to Rocket

Commit:
Echo back one command

Commit:
Echo back multiple commands

Commit:
Edited while loop

Commit:
Add and List

iantsaii committed:
Add and use Task class

iantsaii committed:
Display tasks as Strings

iantsaii committed:
Add ability to mark and unmark tasks

iantsaii committed:
Implement Deadline

iantsaii committed:
Use switch instead of multiple if statements

iantsaii committed:
Implement Todo

iantsaii committed:
Implement Event

iantsaii committed:
Fix minor problem

iantsaii committed:
Implement testing

iantsaii committed:
Implement Exception Handling

iantsaii committed:
Implement delete

iantsaii committed:
Write Java Docs

iantsaii committed:
Fix minor problems

iantsaii committed:
Reformat code

iantsaii committed:
Add test cases

iantsaii committed:
Add test cases

iantsaii committed:
Add ability to save task list

iantsaii committed:
Add ability to retrieve task list from hard disk

iantsaii committed:
Fix extra brackets added to retrieved events

iantsaii committed:
Make function that can convert string to datetime

iantsaii committed:
Make function that can convert dateTime to nicely formatted string

iantsaii committed:
Complete level 8

iantsaii committed:
Complete level 8

iantsaii committed:
Complete level 8

iantsaii committed:
Complete level 8

iantsaii committed:
Complete level 8

iantsaii committed:
Implement more OOP

iantsaii committed:
Add all files to package "rocket"

iantsaii committed:
Merge remote-tracking branch 'origin/add-gradle-support'

iantsaii committed:
Use Gradle

iantsaii committed:
Add Junit tests

iantsaii committed:
Add Jar

iantsaii committed:
Add Java Docs

iantsaii committed:
Edited code to fit coding standard

iantsaii committed:
Add Find Command capability

iantsaii committed:
Merge branch 'branch-A-JavaDoc'

iantsaii committed:
Merge branch 'branch-A-CodingStandard'

iantsaii committed:
Fix FindCommand

lihongguang00 commented on src/main/java/rocket/Command.java:
```diff
@@ -0,0 +1,27 @@
+     * @param ui Displays to the UI if needed.
+     * @param storage Stores and retrieves data from the storage.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
Is this function left intentionally empty?

lawruixi replied:
Perhaps it could be an abstract method instead, if all the subclasses are meant to override it?

lihongguang00 commented on src/main/java/rocket/Parser.java:
```diff
@@ -0,0 +1,91 @@
+            command = fullCommand.substring(0, firstWordIndex);
+            arguments = fullCommand.substring(firstWordIndex + 1);
+        }
+        switch (command) {
```
I like your use of switch here instead of if-else.

lawruixi replied:
Additionally, using braces creates another scope for the variables, and I guess groups the code logically together, although should take care not to accidentally put code outside the braces that might cause bugs or scope issues.

lihongguang00 commented on src/main/java/rocket/Parser.java:
```diff
@@ -0,0 +1,91 @@
+     * @throws RocketException if user enters an invalid command.
+     */
+    public static Command parse(String fullCommand) throws RocketException{
+        String command;
```
I like your use of intuitive variable names that follow java coding standard.

lihongguang00 commented on src/main/java/rocket/Rocket.java:
```diff
@@ -0,0 +1,59 @@
+
+
+
+public class Rocket {
```
I like how your class is organized very clearly. Logic flow is easy to follow.

lihongguang00 commented on src/main/java/rocket/RocketException.java:
```diff
@@ -0,0 +1,7 @@
+package rocket;
+
+public class RocketException extends Exception {
```
I like the way you defined an exception for this task, with additional sub-classes under it to highlight the problem.

lihongguang00 commented on src/main/java/rocket/Storage.java:
```diff
@@ -0,0 +1,93 @@
+     * Create new storage at the filePath specified
+     * @param filePath path to store the file.
+     */
+    public Storage (String filePath){
```
I like this very well structured class with appropriate java doc comments.

lihongguang00 commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,38 @@
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
-
+    ____________________________________________________________
```
I like your use of input to test your program.

lihongguang00 reviewed (commented):
Overall very well structured program, I really like the way you named your variables and the great use of encapsulation and polymorphism. Good job!

iantsaii committed:
Add GUI

lawruixi commented on src/main/java/rocket/AddCommand.java:
```diff
@@ -0,0 +1,26 @@
+package rocket;
+
+public class AddCommand extends Command{
```
```suggestion
public class AddCommand extends Command {
```

lawruixi commented on src/main/java/rocket/TaskList.java:
```diff
@@ -0,0 +1,100 @@
+            }
+            if (deadline.isEmpty()) {
+                throw new RocketIllegalArgumentException("The deadline of a deadline");
+            }
```
I like the use of guard clauses here :D

lawruixi reviewed (commented):
Altogether, code structure looks really well-thought out. Very impressed, and looks good to me.

iantsaii committed:
Add assertions

Unable to ensure that there is a valid file path where the user can save their list of items.

Add assertion to ensure that the file path cannot be empty.

No other methods to check if file path is invalid.

iantsaii committed:
Improve Code Quality

Code is difficult to read and understand

Use Single Level of Abstraction Principle to abstract functions

So that code reviewer can understand the code at a high level of abstraction at first.

iantsaii committed:
Merge pull request #1 from iantsaii/branch-A-Assertions

Add assertions

iantsaii committed:
Merge branch 'master' into branch-A-CodeQuality

iantsaii committed:
Merge pull request #2 from iantsaii/branch-A-CodeQuality

Improve Code Quality

iantsaii committed:
Merge branch 'master' of https://github.com/iantsaii/ip

iantsaii committed:
Merge branch 'master' into branch-A-Assertions

iantsaii committed:
Add case-insensitve searching for BetterSearch

iantsaii committed:
Merge branch 'master' into branch-A-Assertions

iantsaii committed:
Merge pull request #3 from iantsaii/branch-A-Assertions

Branch a assertions

iantsaii committed:
Updated README and screenshot

iantsaii committed:
Merge branch 'master' of https://github.com/iantsaii/ip

iantsaii committed:
Update README.md

iantsaii committed:
Add final changes for IP

iantsaii committed:
Final IP changes
